### PR TITLE
Auto populating current data time field on Simulator UI

### PR DESCRIPTION
**Pull request by RajeevReddyIlavala**

### CODE REVIEW

File: mercury/static/mercury/js/simulator.js [hunk 1]
```diff
@@ -84,6 +84,11 @@ $(function () {
         var suspension_fl = parseInt($('#post-suspension-fl').val());
         var suspension_br = parseInt($('#post-suspension-br').val());
         var suspension_bl = parseInt($('#post-suspension-bl').val());
+        var today = new Date();
+        var date = today.getFullYear()+'-'+("0"+(today.getMonth()+1)).slice(-2)+'-'+("0"+today.getDate()).slice(-2);
```
Comment (de846) [inline mercury/static/mercury/js/simulator.js]: This is really unreadable. I think we need to clean this up somehow.

**Comment by de846** [comment]
@RajeevReddyIlavala - I committed a change here in the GitHub site. Can you take a look and see if its okay with you?

**Comment by de846** [comment]
> I don't think we need a function call in here and 'if' conditions are costly computations. The same functionality can be achieved with slicing which I have done are just string manipulations. I feel like even the lines I wrote are easier to understand as well.

Can you show me how using 'if' is costly for our code?

**Comment by de846** [comment]
I'm even okay using slice instead of `if` if you insist, but putting it all on one line like that with no whitespace is objectively hard to read. We would lose points from readability in the instructor's review.

**Comment by RajeevReddyIlavala** [comment]
> > I don't think we need a function call in here and 'if' conditions are costly computations. The same functionality can be achieved with slicing which I have done are just string manipulations. I feel like even the lines I wrote are easier to understand as well.
> 
> Can you show me how using 'if' is costly for our code?

In general they are expensive especially in a loop, the lines I wrote are not that unreadable.
https://stackoverflow.com/questions/315306/is-if-expensive

https://blogs.sas.com/content/iml/2012/02/13/avoid-unnecessary-if-then-statements-in-loops.html

**Pull request merged**